habitat_baselines/rl/ddppo/policy/resnet_policy.py

```diff
@@ -216,18 +224,26 @@ def __init__(
         normalize_visual_inputs: bool,
         force_blind_policy: bool = False,
         discrete_actions: bool = True,
+        fuse_keys: Optional[List[str]] = None,
     ):
         super().__init__()
 
         self.discrete_actions = discrete_actions
         if discrete_actions:
             self.prev_action_embedding = nn.Embedding(action_space.n + 1, 32)
         else:
-            self.prev_action_embedding = nn.Linear(action_space.n, 32)
+            num_actions = get_num_actions(action_space)
+            self.prev_action_embedding = nn.Linear(num_actions, 32)
 
         self._n_prev_action = 32
         rnn_input_size = self._n_prev_action
 
+        self._fuse_keys = fuse_keys
+        if self._fuse_keys is not None:
+            rnn_input_size += sum(
+                [observation_space.spaces[k].shape[0] for k in self._fuse_keys]
```

Comment:
Could be a generator expression
```suggestion
                (observation_space.spaces[k].shape[0] for k in self._fuse_keys)
```